cmd/oras/internal/option/target.go

```diff
@@ -102,14 +102,13 @@ func (opts *Target) Parse() error {
 		if len(opts.headerFlags) != 0 {
 			return errors.New("custom header flags cannot be used on an OCI image layout target")
 		}
-		return nil
+		_, _, err := opts.parseOCILayoutReference()
+		return err
 	default:
 		opts.Type = TargetTypeRemote
-		if ref, err := registry.ParseReference(opts.RawReference); err != nil {
-			return err
-		} else if ref.Registry == "" || ref.Repository == "" {
+		if _, err := registry.ParseReference(opts.RawReference); err != nil {
 			return &oerrors.Error{
-				Err:            fmt.Errorf("%q is an invalid reference", opts.RawReference),
+				Err:            fmt.Errorf("%w: %q is an invalid reference", err, opts.RawReference),
```

Comment:
Since `ErrInvalidReference` is already `errors.New("invalid reference")`, how about 
```suggestion
				Err:            fmt.Errorf("%w: %q", err, opts.RawReference),
```